engine/src/main/java/org/terasology/logic/console/ui/MiniChatOverlay.java

```diff
@@ -62,13 +65,21 @@ public void initialise() {
             @Override
             public String get() {
                 Iterable<Message> msgs = console.getMessages(CoreMessageType.CHAT, CoreMessageType.NOTIFICATION);
-                String last = "";
+                ArrayList<Message> msgsList = new ArrayList<>();
+                StringBuilder messageHistory = new StringBuilder();
 
                 for (Message msg : msgs) {
-                    last = msg.getMessage();
+                    msgsList.add(msg);
                 }
 
-                return last;
+                //checks if number of messages > MAX_LINES, else starts from 0
+                for (int i = msgsList.size() >= MAX_LINES ? msgsList.size() - MAX_LINES : 0; i < msgsList.size(); i++) {
+                    messageHistory.append(msgsList.get(i).getMessage());
+                    if (i < msgsList.size() - 1) {
+                        messageHistory.append(Console.NEW_LINE);
+                    }
+                }
+                return messageHistory.toString();
```

Comment:
Not sure what is on the classpath, but this looks like a good use for [EvictingQueue](http://google.github.io/guava/releases/snapshot/api/docs/com/google/common/collect/EvictingQueue.html) or [CircularFifoQueue](http://commons.apache.org/proper/commons-collections/javadocs/api-release/org/apache/commons/collections4/queue/CircularFifoQueue.html).
Or even better: Build the string while iterating over the messageList, if you use the counter anyway.